common/models/user.js

```diff
@@ -360,21 +367,34 @@ module.exports = function(User) {
     // Email model
     var Email = options.mailer || this.constructor.email || loopback.getModelByType(loopback.Email);
 
-    crypto.randomBytes(64, function(err, buf) {
-      if (err) {
-        fn(err);
-      } else {
-        user.verificationToken = buf.toString('hex');
-        user.save(function(err) {
-          if (err) {
-            fn(err);
-          } else {
-            sendEmail(user);
-          }
-        });
-      }
+    // Set a default token generation function if one is nto provided
+    var tokenGenerator = options.tokenGenerator || defaultTokenGenerator;
+
+    tokenGenerator(user, function(token) {
+      user.verificationToken = token;
+      user.save(function(err) {
+        if (err) {
+          fn(err);
+        } else {
+          sendEmail(user);
+        }
+      });
     });
 
+    /**
+     * A default verification token generator which accepts a user and callback.
+     * This one uses the crypto library and 64 random bytes (converted to hex)
+     * for the token.
+     *
+     * @options {Object} user The user this token is being generated for
+     * @options {Function} tokenCallback The generator will pass back the new token with this function call
+     */
+    function defaultTokenGenerator(user, tokenCallback) {
```

Comment:
The other option is to have this method  on `User.prototype`. Then it can be easily overridden without passing the generator function in options for each call.